diff --git a/src/init_us_2_no_phase.f90 b/src/init_us_2_no_phase.f90
index 7b69d32..7a1ff61 100644
--- a/src/init_us_2_no_phase.f90
+++ b/src/init_us_2_no_phase.f90
@@ -24,7 +24,6 @@ subroutine init_us_2_no_phase (npw_, igk_, q_, vkb_)
   USE uspp,       ONLY : nkb, nhtol, nhtolm, indv
   USE uspp_param, ONLY : upf, lmaxkb, nhm, nh
   USE gipaw_module, ONLY : spline_ps, tab_d2y
-  USE ieee_arithmetic
   !
   implicit none
   !
@@ -117,7 +116,7 @@ subroutine init_us_2_no_phase (npw_, igk_, q_, vkb_)
               lm =nhtolm (ih, nt)
               do ig = 1, npw_
                  vkb1 (ig,ih) = ylm (ig, lm) * vq (ig)
-                 if (ieee_is_nan(vq(ig))) STOP
+                 if (isnan(vq(ig))) STOP
               enddo
            endif
         enddo
